lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
Isn't `collection == self` already doing this?  See https://github.com/projecthydra/active_fedora/blob/master/lib/active_fedora/core.rb#L68-L72